tests/components/compensation/test_sensor.py

```diff
@@ -41,8 +86,17 @@ async def test_linear_state(hass):
     state = hass.states.get(expected_entity_id)
     assert state is not None
 
+    ent_reg = entity_registry.async_get(hass)
+    registry_entry = ent_reg.async_get(expected_entity_id)
+
+    # Entity is not registered when unique_id is not configured
+    assert registry_entry is None
```

Comment:
Added assertion when "unique_id" isn't configured. The entity will not be registered and configurable in the UI.